scaling_performance/host_practices.adoc

```diff
@@ -57,42 +57,94 @@ include writing a large amount of log data.
 etcd is a distributed key-value store that {product-title} uses for
 configuration.
 
+Version 3 of etcd brings with it important scalability and performance improvements that reduce CPU, memory, network and disk requirements for any size cluster.  OpenShift 3.5 introduced etcd version 3.1.
+
+Version 3 of etcd implements a backwards compatible storage API that facilitates a two-step migration of the on-disk etcd database.  For migration purposes, the storage mode used by etcd 3.1 in OpenShift 3.5 remained in v2 mode.  As of OpenShift 3.6, new installs will use storage mode v3.  Upgrades from previous versions of OpenShift will NOT automatically migrate data from v2 to v3.  This must be done using supplied playbooks following the documented process here: TODO find link to migrate docs.
+
+In addition to changing the storage mode for new installs to v3, OpenShift 3.6 also begins enforcing "quorum reads" for all OpenShift types.  This is done to ensure that queries against etcd do not return stale data.  In single-node etcd clusters, stale data is not a concern.  In highly available etcd deployments typically found in production clusters, quorum reads ensure valid query results.  Quorum reads force an fsync to guarantee all nodes are in sync.  This behavior is part of the raft algorithm design.
```

Comment:
As per [1], "While etcd 3.0 processes linearized reads through direct raft proposals, chewing precious disk bandwidth and incurring the corresponding latency penalty, linearized reads in 3.1 issue idempotent, fsync-free linearized raft index requests. These index requests are also batched; concurrent linearized reads coalesce into a single index request. The upshot is linearized reads have lower latency and better throughput."

So it is probably incorrect to state that 'Quorum reads force an fsync to guarantee all nodes are in sync' as that does not seem to be the case in etcd 3.1. May be a better idea to refer to [1] for the details on how quorum reads are handled in etcd 3.1?

[1] https://coreos.com/blog/etcd-3-1-announcement.html